### ISSUE TITLE
Existing System Ruby Breaks azure-chef-extension install.sh/enable.sh

### ISSUE THREAD

**Issue by sean-horn** [chef-partners/azure-chef-extension#131]
--- Versions:

OS: RHEL7
azure-chef-extension: latest

--- Problem:

As a customer, I want to be able to have a system ruby installed alongside an embedded chef-client ruby.
Redhat Satellite and other things depend on the system ruby.

When a system ruby is installed, the current version of azure-chef-extension fails while being installed.

It has the following output while manually running `enable.sh` directly on the system as a test, because the gem gets installed in the system ruby gem location, rather than the embedded chef-client gem location. 

```
root@flapvm-azpcc01 Chef.Bootstrap.WindowsAzure.LinuxChefClient-1210.12.102.1000]# ./enable.sh

/var/lib/waagent/Chef.Bootstrap.WindowsAzure.LinuxChefClient-1210.12.102.1000

/opt/chef/embedded/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- chef/azure/commands/enable (LoadError)

from /opt/chef/embedded/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'

from /var/lib/waagent/Chef.Bootstrap.WindowsAzure.LinuxChefClient-1210.12.102.1000/bin/chef-enable.rb:4:in `<main>'
```

This line is actually where the failure is occurring https://github.com/chef-partners/azure-chef-extension/blob/9e9d7b831f5dbb7d5d3c13303c65f0fa1b1b0796/ChefExtensionHandler/bin/chef-enable.rb#L4

--- Possible fix (There is likely a better way to fix this, please do):

1. Add full path /opt/chef/embedded/bin to have azure-chef-extension gem installed in the
   embedded chef-client gems in chef-install.sh https://github.com/chef-partners/azure-chef-extension/blob/9e9d7b831f5dbb7d5d3c13303c65f0fa1b1b0796/ChefExtensionHandler/bin/chef-install.sh#L18

2. Add full path /opt/chef/embedded/bin to chef-client embedded ruby in chef-enable.sh  https://github.com/chef-partners/azure-chef-extension/blob/9e9d7b831f5dbb7d5d3c13303c65f0fa1b1b0796/ChefExtensionHandler/enable.sh#L17

When these two changes are made, even with a system ruby present, we expect 

1. That the azure-chef-extension gem will be installed in the correct embedded gem location
2. The correct embedded ruby will be called, so that the `require` in bin/chef-enable.rb will not fail

**Comment by jjasghar** [comment]
@sean-horn i think this https://github.com/chef-partners/azure-chef-extension/pull/132 fixes it. @russellseymour can you add your thoughts here?

**Comment by russellseymour** [comment]
@sean-horn We took option 2 so that the system ruby is not before the Chef ruby.

We did think to just protect for Redhat based distros but then as Azure lets you install an extension at any time in the life of the machine there was a chance that an Ubuntu machine, for example, may have ruby installed through the package manager.

This would have then meant the same problem on other distros.

**Comment by sean-horn** [comment]
I believe https://github.com/chef-partners/azure-chef-extension/blob/master/ChefExtensionHandler/bin/chef-install.sh#L209 also needs to have the PATH setting changed, otherwise if there is a system ruby installed, the gem location referred to will be that one, rather than the chef-client embedded gem location.

**Comment by kevinreedy** [comment]
@NimishaS Thanks for getting the fix merged! Do you plan on doing a release/publish soon?

**Comment by NimishaS** [comment]
@kevinreedy , this fix has been published in extension `1210.12.103.1000`.
